### PR TITLE
Added mapping and sequence setters

### PR DESCRIPTION
**Pull request by jonogilmour**

### CODE REVIEW

File: set_test.go [hunk 1]
```diff
@@ -44,3 +44,92 @@ key: [1, "hi"]
...
+	// key: [1, "hi"]
+	// h:
+	//     - abc
+	//     - 123
+	// new:
+	//     - yello
+	//     - true
+	//     - 123
+	//     - hello
+}
```
[the remark this answers is not in the document]
Comment (jonogilmour) [reply set_test.go:81]: it tests the output comment at the end is what is output. Some kind of magic.

File: set.go [hunk 2]
```diff
@@ -140,3 +140,53 @@ func SetInt(node *goyaml.Node, path Path, value int64) (valueNode *goyaml.Node,
...
+//
+// The node representing the sequence is returned.
+// This can be used to control the style/comments in the output YAML.
+func SetSequence(node *goyaml.Node, path Path, content []any) (valueNode *goyaml.Node, changed bool, err error) {
+	var contents []*goyaml.Node
+	for _, c := range content {
+		// Depending on the type of each value in content, add a yaml node representation of it to contents
+		switch v := c.(type) {
+		default:
+			err = fmt.Errorf("unsupported type %T in sequence", v)
```
[the remark this answers is not in the document]
Comment (jonogilmour) [reply set.go:157]: good point! Added

File: set_test.go [hunk 3]
```diff
@@ -44,3 +44,71 @@ key: [1, "hi"]
...
+# Comment that should be preserved.
+key: [1, "hi"]
+h:
+- abc
+- 123
+`), docNode)
+	node := docNode.Content[0]
+
+	// Edit.
+	_, _ = SetScalar(node, MustParsePath(".new"), "some content")
```
Comment (brekelj1) [inline set_test.go:61]: Add two more test cases:
- 1 for if a value already exists at the specified path, but has a different value (assert changed=true).
- another for if a value already exists at the specified path, but has equal value (assert changed=false).

File: set.go [hunk 4]
```diff
@@ -140,3 +140,49 @@ func SetInt(node *goyaml.Node, path Path, value int64) (valueNode *goyaml.Node,
...
+//
+// path is the path within the node to set the value at.
+//
+// value is the scalar value to set, eg. "hello".
+//
+// Returns a bool showing whether or not the value set updated an existing value (ie. value was changed).
+func SetScalar(node *goyaml.Node, path Path, value any) (changed bool, err error) {
+	n, err := plumbing.MakeScalar(value)
+	if err == nil {
+		_, changed, err = set(node, path, nil, func() *goyaml.Node {
```
Comment (brekelj1) [inline set.go:156]: Bugfix. Add test cases for this: https://github.com/jbrekelmans/go-edityaml/pull/5/files#r1577151228
```suggestion
		_, changed, err = set(node, path, value, func() *goyaml.Node {
```

File: set.go [hunk 5]
```diff
@@ -140,3 +140,49 @@ func SetInt(node *goyaml.Node, path Path, value int64) (valueNode *goyaml.Node,
...
+//
+//	value := &goyaml.Node{
+//		Kind: goyaml.MappingNode,
+//		Tag:  "!!map",
+//	}
+//
+// err := Set(node, ".x.y", value)
+// ...
+func Set(node *goyaml.Node, path Path, value *goyaml.Node) (addedNode *goyaml.Node, err error) {
+	addedNode, _, err = set(node, path, nil, func() *goyaml.Node {
```
Comment (brekelj1) [inline set.go:184]: We have to refactor `set` slightly as it hardcodes comparing an existing value with `null` now, and skips setting if existing value is `null`.

**Review by brekelj1 — APPROVED** [review]

**Pull request merged**